cmd/deploy/validate.go

```diff
@@ -23,14 +23,23 @@ type envVar struct {
 	value string
 }
 
-func validateAndSet(variables []string, setEnv func(key, value string) error) error {
-	envVars, err := parse(variables)
+func validateAndSetVariables(opts *Options, setEnv func(key, value string) error) error {
+	cleanEscapedVariables(opts)
+	envVars, err := parse(opts.Variables)
 	if err != nil {
 		return err
 	}
 	return setOptionVarsAsEnvs(envVars, setEnv)
 }
 
+func cleanEscapedVariables(opts *Options) {
+	cleanedVariables := []string{}
+	for _, v := range opts.Variables {
+		cleanedVariables = append(cleanedVariables, strings.ReplaceAll(v, "\"", ""))
```

Comment:
How can I reproduce this case? When we do this for the name we don't have to do it

Reply:
you can check the e2e test for it. We need to escape values in the installer in order to forward variables using `exec.Command`. If we do not, you will not be able to use the proper value, instead of that, you will use something like: `\"a value\"`. In the following example, config map name cannot contain `\`.
````yaml
deploy:
  - kubectl create cm $OKTETO_DEPENDENCY_TEST_VARIABLE_CMNAME --from-literal=key1="$OKTETO_DEPENDENCY_TEST_VARIABLE_MY_DYNAMIC_APP_ENV"
dependencies:
  test:
    repository: https://github.com/okteto/go-getting-started
    branch: generating-dynamic-envs
    variables:
      CMNAME: test
    wait: true
````

Reply:
It's weird because If I work directly with the terminal cobra doesn't add the "" but if I executed it with exec it does 🤔.

Maybe we can try to use `--` to let the `cmd.Exec` that is a single argument. 

The problem with the code above is that if a user wants to add a variable that contains a double quote it will get removed...